src/plugins/data/public/antlr/opensearch_sql/code_completion.test.ts

```diff
@@ -0,0 +1,149 @@
+    } as IDataPluginServices;
+
+    const mockPosition = {
+      lineNumber: 1,
+      column: 1,
+    } as monaco.Position;
+
+    const getSimpleSuggestions = async (
+      query: string,
+      position: monaco.Position = new monaco.Position(1, query.length + 1)
+    ) => {
+      return getSuggestions({
+        query,
+        indexPattern: mockIndexPattern,
+        position,
+        language: 'SQL',
+        selectionStart: 0,
+        selectionEnd: 0,
+        services: mockServices,
+      });
+    };
+
+    const checkSuggestionsContain = (
+      result: QuerySuggestion[],
+      expected: Partial<QuerySuggestion>
+    ) => {
+      expect(
+        result.some(
+          (suggestion) => suggestion.text === expected.text && suggestion.type === expected.type
+        )
+      ).toBeTruthy();
+    };
+
+    it('should return empty array when required parameters are missing', async () => {
+      const result = await getSuggestions({
+        query: '',
+        indexPattern: (null as unknown) as IndexPattern,
+        position: mockPosition,
+        language: 'SQL',
+        selectionStart: 0,
+        selectionEnd: 0,
+        services: (null as unknown) as IDataPluginServices,
+      });
+
+      expect(result).toEqual([]);
+    });
+
+    it('should suggest table name when suggestViewsOrTables is true', async () => {
+      const result = await getSimpleSuggestions('SELECT * FROM ');
+
+      checkSuggestionsContain(result, {
+        text: 'test-index',
+        type: monaco.languages.CompletionItemKind.Struct,
+      });
+    });
+
+    it('should suggest columns when suggestColumns is true', async () => {
+      const result = await getSimpleSuggestions('SELECT * FROM test-index WHERE ');
+
+      checkSuggestionsContain(result, {
+        text: 'field1',
+        type: monaco.languages.CompletionItemKind.Field,
+      });
+    });
+
+    it('should suggest EQ for column value predicate', async () => {
+      const result = await getSimpleSuggestions('SELECT * FROM test-index WHERE field1 ');
+
+      checkSuggestionsContain(result, {
+        text: '=',
+        type: monaco.languages.CompletionItemKind.Operator,
+      });
+    });
+
+    it('should suggest LPAREN for column value predicate', async () => {
+      const result = await getSimpleSuggestions('SELECT * FROM test-index WHERE field1 IN ');
+
+      checkSuggestionsContain(result, {
+        text: '(',
+        type: monaco.languages.CompletionItemKind.Keyword,
+      });
+    });
+
+    it('should suggest END_IN_TERMs for column value predicate', async () => {
+      const result = await getSimpleSuggestions(
+        "SELECT * FROM test-index WHERE field1 IN ( 'value' "
+      );
+
+      [',', ')'].forEach((term) => {
+        checkSuggestionsContain(result, {
+          text: term,
+          type: monaco.languages.CompletionItemKind.Keyword,
+        });
+      });
+    });
+
+    it.skip('should suggest aggregate functions when appropriate', async () => {
```

Comment:
NOTE: This test is skipped for now due to the working functionality requiring a new 'rerun and combine' functionality, where preferred rules are removed one at a time and then the completion candidate process is run without that rule. 
This new system is already written in this PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/9120 
Decoupling it from the rest of that PR would be too much unnecessary effort.